x/nexus/keeper/chain.go

```diff
@@ -248,3 +223,75 @@ func (k Keeper) setChainByNativeAsset(ctx sdk.Context, asset string, chain expor
 func (k Keeper) GetChainByNativeAsset(ctx sdk.Context, asset string) (chain exported.Chain, ok bool) {
 	return chain, k.getStore(ctx).Get(chainByNativeAssetPrefix.Append(utils.LowerCaseKey(asset)), &chain)
 }
+
+type ChainKeeper struct {
+	k      Keeper
+	states map[string]*cached.Cached[types.ChainState]
+	dirty  map[string]bool
+}
+
+func NewChainKeeper(k Keeper) *ChainKeeper {
+	return &ChainKeeper{
+		k:      k,
+		dirty:  map[string]bool{},
+		states: map[string]*cached.Cached[types.ChainState]{},
+	}
+}
+
+// MarkChainMaintainerMissingVote marks the given chain maintainer for missing vote of a poll
+func (k *ChainKeeper) MarkMissingVote(ctx sdk.Context, chain exported.Chain, address sdk.ValAddress, missingVote bool) {
+	k.markMisbehave(ctx, chain, address, missingVote, func(maintainerState *types.MaintainerState) *utils.Bitmap {
+		return &maintainerState.MissingVotes
```

Comment:
Doesn't need to be included in this PR, but probably should just add getters for MissingVotes, and IncorrectVotes, and then the getters can be the function we pass in?